src/isar/robot/robot_status.py

```diff
@@ -35,10 +34,10 @@ def _is_ready_to_poll_for_status(self) -> bool:
         )
 
     def run(self):
-        while True:
-            if self.signal_thread_quitting.is_set():
-                break
+        if self.signal_thread_quitting.is_set():
+            return
 
+        while not self.signal_thread_quitting.wait(0.001):
```

Comment:
It was necessary to check both before the while loop and in the condition for one of the tests. I think it was because the quitting signal was sent even before the run function was entered